core/templating/datasource_sql_over_csv.go

```diff
@@ -0,0 +1,375 @@
+		return headers, nil
+	}
+	columns := strings.Split(columnsPart, ",")
+	for i, column := range columns {
+		if !stringExists(headers, strings.TrimSpace(column)) {
+			return nil, errors.New("invalid column provided: " + strings.TrimSpace(column))
+		}
+		columns[i] = strings.TrimSpace(column)
+	}
+	return columns, nil
+}
+
+func stringExists(slice []string, target string) bool {
+	for _, s := range slice {
+		if s == target {
+			return true
+		}
+	}
+	return false
+}
+
+// TrimQuotes trims matching single or double quotes from the outer edges of a string.
+func trimQuotes(s string) string {
+	// Check if the string length is less than 2, in which case there's nothing to trim.
+	if len(s) < 2 {
+		return s
+	}
+
+	// Get the first and last characters of the string.
+	firstChar := s[0]
+	lastChar := s[len(s)-1]
+
+	// Check if both the first and last characters are matching quotes.
+	if (firstChar == '\'' || firstChar == '"') && firstChar == lastChar {
+		return s[1 : len(s)-1]
+	}
+
+	// If no trimming is necessary, return the original string.
+	return s
+}
+
+// parseSetClauses parses the SET part of an UPDATE query
+func parseSetClauses(setPart string) map[string]string {
+	setClauses := make(map[string]string)
+	parts := strings.Split(setPart, ",")
+	for _, part := range parts {
+		keyValue := strings.Split(strings.TrimSpace(part), "=")
+		if len(keyValue) == 2 {
+			setClauses[strings.TrimSpace(keyValue[0])] = trimQuotes(strings.TrimSpace(keyValue[1]))
+		}
+	}
+	return setClauses
+}
+
+// parseConditions parses the WHERE part of the query into a slice of Conditions and returns an error if any issues are found.
+func parseConditions(wherePart string) ([]Condition, error) {
+	conditions := []Condition{}
+
+	conditionRegex := regexp.MustCompile(`(\w+)\s*(==|!=|<=|>=|<|>)\s*'([^']*)'`)
+	conditionMatches := conditionRegex.FindAllStringSubmatch(wherePart, -1)
+
+	if len(conditionMatches) == 0 {
+		return conditions, nil
+	}
+
+	for _, match := range conditionMatches {
+		if len(match) != 4 {
+			return nil, errors.New("invalid condition format")
+		}
+		conditions = append(conditions, Condition{
+			Column:   match[1],
+			Operator: match[2],
+			Value:    match[3],
+		})
+	}
+
+	return conditions, nil
+}
+
+// ExecuteSelectQuery executes a SELECT query and returns the results as a slice of RowMaps
+func executeSqlSelectQuery(data *[][]string, query SQLStatement) []RowMap {
+	headers := (*data)[0] // First row as header
+	results := []RowMap{}
+	for _, row := range (*data)[1:] {
+		rowMap := mapRow(headers, row)
+		if matchesConditions(rowMap, query.Conditions) {
+			results = append(results, projectRow(rowMap, query.Columns))
+		}
+	}
+	return results
+}
+
+// ExecuteUpdateQuery executes an UPDATE query and modifies the data in-place
+func executeSqlUpdateCommand(data *[][]string, query SQLStatement) []RowMap {
+	if len(*data) < 2 {
+		log.Error("no data available to update")
+		return []RowMap{
+			{
+				"rowsAffected": "0",
+			},
+		}
```

Comment:
I was thinking of returning an integer only, `UPDATE` / `DELETE` in SQL doesn't return a result set, so there is no need to wrap it in the row record. This special `"rowsAffected": "0"` will get returned to the templated response too which I think it's unnecessary.

I was thinking that  `executeSqlUpdateCommand` returns an int, then the caller `csvSqlCommand` will log the counts, and return nil for the result set.

Reply:
I have updated update and delete to log rows and return nil